app/controllers/admin/users/list.js

```diff
@@ -1,108 +1,116 @@
-      title            : 'Email',
-      disableSorting   : true,
-      disableFiltering : true
-    },
-    {
-      propertyName     : 'status',
-      title            : 'Status',
-      template         : 'components/ui-table/cell/admin/users/cell-status',
-      disableSorting   : true,
-      disableFiltering : true
-    },
-    {
-      propertyName     : 'system-roles',
-      title            : 'System Roles',
-      template         : 'components/ui-table/cell/admin/users/cell-system-roles',
-      disableSorting   : true,
-      disableFiltering : true
-    },
-    {
-      propertyName     : 'event-roles',
-      title            : 'Event Roles',
-      template         : 'components/ui-table/cell/admin/users/cell-event-roles',
-      disableSorting   : true,
-      disableFiltering : true
-    },
-    {
-      propertyName     : 'user-links',
-      title            : 'User Links',
-      template         : 'components/ui-table/cell/admin/users/cell-user-links',
-      disableSorting   : true,
-      disableFiltering : true
-    },
-    {
-      propertyName : 'created-at',
-      title        : 'Member Since',
-      template     : 'components/ui-table/cell/admin/users/cell-created-at'
-    },
-    {
-      propertyName     : 'lastAccessedAt',
-      title            : 'Last Accessed',
-      template         : 'components/ui-table/cell/cell-simple-date',
-      dateFormat       : 'MMMM DD, YYYY - hh:mm A',
-      disableSorting   : true,
-      disableFiltering : true
-    },
-    {
-      template         : 'components/ui-table/cell/admin/users/cell-actions',
-      title            : 'Actions',
-      disableSorting   : true,
-      disableFiltering : true
+import { computed, action } from '@ember/object';
+import { or } from '@ember/object/computed';
+import EmberTableControllerMixin from 'open-event-frontend/mixins/ember-table-controller';
+
+
+export default class extends Controller.extend(EmberTableControllerMixin) {
+  @or('authManager.currentUser.isSuperAdmin', 'authManager.currentUser.isAdmin') hasRestorePrivileges;
+
+  @computed()
+  get columns() {
+    return [
+      {
+        name          : 'Name',
+        valuePath     : null,
+        cellComponent : 'ui-table/cell/admin/users/cell-first-name',
+        options       : {
+          hasRestorePrivileges: this.hasRestorePrivileges
+        },
+        actions: {
+          moveToUserDetails : this.moveToUserDetails.bind(this),
+          deleteUser        : this.deleteUser.bind(this),
+          openEditModal     : this.openEditModal.bind(this),
+          restoreUser       : this.restoreUser.bind(this)
+        }
+
+      },
+      {
+        name      : 'Email',
+        valuePath : 'email'
+
+
+      },
+      {
+        name          : 'Status',
+        valuePath     : 'status',
+        cellComponent : 'ui-table/cell/admin/users/cell-status'
+
+
+      },
+      {
+        name          : 'System Roles',
+        valuePath     : null,
```

Comment:
to pass the entire record @niranjan94

Reply:
Why entire record ? Couldn't you pass only the properties you need ?

Reply:
@niranjan94 Okay, I can provide around 10 extra valuePaths for the properties it requires, but for the column which has buttons, I do need the entire record in the cell component because it's action uses record.destroyRecord(). Is that okay, or the action needs to be refactored?

Reply:
> I can provide around 10 extra valuePaths for the properties it requires

I think that's okay. That way we explicitly show what all data that cell requires.

> I do need the entire record in the cell component because it's action uses

Not at all. You only need an `id`. With that ID you can just do a peekRecord in the action to get the full record from the store. This will be a simple, fast synchronous operation. 

--

Pass in just the data your cells require. Nothing more.